src/components/FeatureViews/AdminPortal/PanelManager.js

```diff
@@ -0,0 +1,100 @@
+import React, {useEffect, useMemo, useState} from "react";
+import * as adminActions from "../../../actions/adminActions";
+import "./AdminPortal.scss";
+import {connect, useDispatch, useSelector} from "react-redux";
+import {bindActionCreators} from "redux";
+
+import Grid from "@material-ui/core/Grid";
+import {Button, Typography} from "@material-ui/core";
+
+function PanelManager(props) {
+/**
+ * Fetches, configures, and displays a grid-structured panel for a given 
+ * set in the store.
+ * 
+ * 
+ * Example: 
+ * { TODO}
+ * 
+ * @prop {array} operations List of operation types (i.e ["READ", "UPDATE" ) => options are: READ, UPDATE, DELETE 
+ */
+    const dispatch = useDispatch();
+    const api = useMemo(
+        () => ({
+            ...bindActionCreators(adminActions, dispatch),
+        }),
+        [dispatch]
+    );
+
+    let [records, setRecords] = useState({});
+
+    const addDefaults = (fields) => props.fields.map((field) => {
+        //Add defaults field properties to the props.fields object
+        Object.keys(defaults).forEach((key) => {
+            if(field.hasOwnProperty(key)) {
+                //Do nothing, property is already set
+            } else {
+                field[key] = defaults[key];
+            }
+        });
+        return field;
+    });
+
+    const defaults = {
+        "editable": "false",
+        "align": "left"
+    };
+
+
+    const generateGrid = () => {
+        /**
+         * Generates header with fields
+         */
+        const fieldsWithDefaults = addDefaults(props.fields);
+        const headerElements = [];
+        const records = [];
+
+        for (const [index, value] of fieldsWithDefaults.entries()) {
```

Comment:
to use better JS, you can use the .forEach() function

example: 
```
fieldsWithDefaults.entries.forEach( ( [index, value] ) => { ... } ); 
```

Reply:
In fact, since you are trying to create a new array based on an existing one, you can use [`.map()`](https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Global_Objects/Array/map):

    const headerElements = fieldsWithDefaults.map((value, index) => /* JSX */);

which can also be directly chained to the `addDefaults` call:

    const headerElements = addDefaults(props.fields).map((value, index) => /* JSX */);